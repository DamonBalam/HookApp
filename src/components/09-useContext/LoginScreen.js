import React, { useContext } from 'react'
import { UserContext } from './UserContext';

export const LoginScreen = () => {

    // 1. obtener la referencia del context
    const  { setUser } = useContext(UserContext);

    // 2. setUser
    
    return (
        <div>
            <h1>LoginScreen</h1>
            <hr />

            <button
                className="btn btn-primary"
                onClick={ ()=> setUser({id:1234, name:'Arturo'}) }
            >
                Login
            </button>
        </div>
    );
}
