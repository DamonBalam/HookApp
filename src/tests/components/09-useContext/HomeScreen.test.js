import { mount, shallow } from "enzyme";
import { HomeScreen } from "../../../components/09-useContext/HomeScreen";
import { UserContext } from "../../../components/09-useContext/UserContext";

describe('Pruebas en el HomeScreen', () => {
    
    const user = {
        name:'Arturo',
        email: 'damonbalam@gmail.com'
    }

    const setUser = ()=> {};


    const wrapper = mount( 
        <UserContext.Provider value={{user}}>
            <HomeScreen/>
        </UserContext.Provider>
    );

    test('Debe de mostrarse correctamente el <HomeScreen />', () => {

        expect( wrapper ).toMatchSnapshot();

    });
    
});
