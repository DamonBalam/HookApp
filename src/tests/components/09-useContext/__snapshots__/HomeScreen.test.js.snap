// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el HomeScreen Debe de mostrarse correctamente el <HomeScreen /> 1`] = `
<div>
  <h1>
    HomeScreen
  </h1>
  <hr />
  <pre>
    {
   "name": "Arturo",
   "email": "damonbalam@gmail.com"
}
  </pre>
</div>
`;
